scripts/full-workflow.sh

```diff
@@ -0,0 +1,363 @@
+    task Model Training Complete
+}
+
+test_phased_train() {
+    task Train the model with LAB multi-phase training
+
+    # linter wants data_path and other variables declared then assigned separately.
+    # 'Declare and assign separately to avoid masking return values' <-- error.
+    local data_path
+    data_path=$(find "${DATA_HOME}"/instructlab/datasets -name 'messages_*' | head -n 1)
+
+    local model_path
+    model_path="${CACHE_HOME}/instructlab/models/${GRANITE_SAFETENSOR_REPO}"
+
+    # general training args - not the same as the global TRAIN_ARGS
+    local train_args
+    # general phased training args
+    train_args+=(
+        "--phased-training"
+        "--skip-user-confirm"
+        "--phased-mt-bench-judge-dir=${model_path}" # uses base model
+    )
+
+    # phase 1 args 
+    train_args+=(
+        "--phased-phase1-data=${data_path}"
+    )
+
+    # phase 2 args 
+    train_args+=(
+        "--phased-phase2-data=${data_path}"
+    )
+
+    ilab model train "${train_args[@]}"
+    # final best model is written to stdout.
+    task Multi-phase Training Complete
+}
+
+test_evaluate() {
+    task Evaluate the model
+    
+    local model_path
+    local base_model_path
+    local judge_model_path
+    local mmlu_branch_tasks_dir
+
+    model_path=${TRAINED_MODEL_PATH}
+    base_model_path="${CACHE_HOME}/instructlab/models/instructlab/granite-7b-lab"
+    judge_model_path="${CACHE_HOME}/instructlab/models/${PROMETHEUS_7B_MODEL}"
+    mmlu_branch_tasks_dir=$(find "${DATA_HOME}"/instructlab/datasets -name 'node_datasets_*' | head -n 1)
+
+    export INSTRUCTLAB_EVAL_MMLU_MIN_TASKS=true
+    export HF_DATASETS_TRUST_REMOTE_CODE=true
+    task Running MMLU
+    ilab model evaluate --model "${model_path}" --benchmark mmlu
+    task Running MMLU_BRANCH
+    ilab model evaluate --model "${model_path}" --benchmark mmlu_branch --tasks-dir "${mmlu_branch_tasks_dir}" --base-model "${base_model_path}"
+
+    export INSTRUCTLAB_EVAL_FIRST_N_QUESTIONS=20
+    task Running MT_Bench
+    ilab model evaluate --model "${model_path}" --judge-model "${judge_model_path}" --enable-serving-output --benchmark mt_bench
+    task Running MT_Bench_Branch
+    cd "${DATA_HOME}/instructlab/taxonomy"
+    git branch rc
+    ilab model evaluate \
+    --model "${model_path}" \
+    --judge-model "${judge_model_path}" \
+    --branch rc --base-branch main \
+    --base-model "${base_model_path}" \
+    --enable-serving-output \
+    --benchmark mt_bench_branch \
+    --taxonomy-path "${DATA_HOME}/instructlab/taxonomy"
+
+    task Evaluation Complete
+}
+
+test_config_show() {
+    task Output active config
+    ilab config show
+    task Config show Complete
+}
+
+test_exec() {
+    # The list of actual tests to run through in workflow order
+    test_smoke
+    test_init
+    test_config_show
+    test_download
+    test_list
+
+    test_taxonomy
+    test_generate
+    test_train
+    #test_phased_train
```

Comment:
Note: I'm working on trying to get phased training to run. In the mean time, I'd like to merge the PR as is.